File: ACM International Collegiate Programming Contest, Arabella Collegiate Programming Contest (2018)/C.cc
#include <iostream>
#include <iomanip>
#include <cmath>

int T;
long double a, b, c, x;

long double dis(long double a, long double b) {
	return std::sqrt(a*a+b*b);
}

int main(int argc, char *argv[]) {  
	std::cin >> T;
	while (T--) {
		std::cin >> a >> b >> c >> x;
		long double ans = 0.0;
		long double xx = b*x/100.0, yy = a*x/100.0;
		ans += dis(b+c+c+b-xx, a-yy);
		ans += dis(xx, yy);
		ans += dis(a, b+c+c);
		std::cout << std::fixed << std::setprecision(12) << ans << '\n';
	}
}

File: ACM International Collegiate Programming Contest, Arabella Collegiate Programming Contest (2018)/C.py
import math

T = int(input())

for i in range(0, T):
	a, b, c, x = map(float, input().split(' '))
	xx = b*x/100.0
	yy = a*x/100.0
	ans = math.sqrt((b+c+c-xx)*(b+c+c-xx)+(a-yy)*(a-yy))
	ans = ans + math.sqrt(xx*xx+yy*yy)
	ans = ans + math.sqrt(a*a+(b+c+c)*(b+c+c))
	print(ans)
